src/location.cpp

```diff
@@ -32,14 +42,21 @@ bool is_github_shorthand(std::string_view s) { return github::is_shorthand(s); }
 
 bool is_github_repo(std::string_view s) { return is_github(s) && !is_archive(s); }
 
+std::string_view strip_github_url_to_shorthand(std::string_view github_url) {
+   if(github_url.size() <= github_com.size())
+      throw std::out_of_range("URL passed: "+  std::string(github_url) + " should have minimum size of " + std::to_string(github_com.size() + 1));
```

Comment:
More specific sanity checking? 
```suggestion
   if(github_url.find(github_com) != 0)
        throw std::out_of_range("URL passed: "+  std::string(github_url) + " should begin with: " + github_com);
```

Reply:
merged previous `normalize` and `strip_github_url_to_shorthand` into one function: `strip_github_com`